src/main/java/graphql/execution/ExecutionPath.java

```diff
@@ -144,12 +148,18 @@ public ExecutionPath segment(String segment) {
      * Takes the current path and adds a new segment to it, returning a new path
      *
      * @param segment the int path segment to add
+     *
      * @return a new path containing that segment
      */
     public ExecutionPath segment(int segment) {
         return new ExecutionPath(this, new IntPathSegment(segment));
     }
 
+    public ExecutionPath sibling(String siblingField) {
+        Assert.assertTrue(!ROOT_PATH.equals(this), "You MUST not call this with the root path");
+        return new ExecutionPath(this.parent, new StringPathSegment(siblingField));
+    }
+
```

Comment:
This is not needed for this PR technically.  But I need it for another so bear with me.

It allows side ways movements of paths

    hero/name --> hero/id